rotkehlchen/chain/ethereum/decoding/pickle/decoder.py

```diff
@@ -0,0 +1,95 @@
+    tx_log: EthereumTxReceiptLog,
+    transaction: EthereumTransaction,
+    event: HistoryBaseEntry,
+    action_items: List[ActionItem],  # pylint: disable=unused-argument
+) -> bool:
+    """Enrich tranfer transactions to address for jar deposits and withdrawals"""
+    if not (
+        hex_or_bytes_to_address(tx_log.topics[2]) in PICKLE_CONTRACTS or
+        hex_or_bytes_to_address(tx_log.topics[1]) in PICKLE_CONTRACTS or
+        tx_log.address in PICKLE_CONTRACTS
+    ):
+        return False
+
+    # Deposit
+    if (
+        event.event_type == HistoryEventType.SPEND and
+        event.event_subtype == HistoryEventSubType.NONE and
+        event.location_label == transaction.from_address and
+        hex_or_bytes_to_address(tx_log.topics[2]) in PICKLE_CONTRACTS
+    ):
+        if EthereumToken(tx_log.address) != event.asset:
+            return True
+        amount_raw = hex_or_bytes_to_int(tx_log.data)
+        amount = asset_normalized_value(amount=amount_raw, asset=event.asset)
+        if event.balance.amount == amount:  # noqa: E501
+            event.event_type = HistoryEventType.STAKING
+            event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+            event.counterparty = 'pickle finance'
+            event.notes = f'Deposit {event.balance.amount} {event.asset.symbol} in pickle contract'  # noqa: E501
+    elif (
+        event.event_type == HistoryEventType.RECEIVE and
+        event.event_subtype == HistoryEventSubType.NONE and
+        tx_log.address in PICKLE_CONTRACTS
+    ):
+        amount_raw = hex_or_bytes_to_int(tx_log.data)
+        amount = asset_normalized_value(amount=amount_raw, asset=event.asset)
+        if event.balance.amount == amount:  # noqa: E501
+            event.event_type = HistoryEventType.STAKING
+            event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+            event.counterparty = 'pickle finance'
+            event.notes = f'Recive {event.balance.amount} {event.asset.symbol} after depositing in pickle contract'  # noqa: E501
+    # Withdraw
+    elif (
+        event.event_type == HistoryEventType.SPEND and
+        event.event_subtype == HistoryEventSubType.NONE and
+        event.location_label == transaction.from_address and
+        hex_or_bytes_to_address(tx_log.topics[2]) == ZERO_ADDRESS and
+        hex_or_bytes_to_address(tx_log.topics[1]) in transaction.from_address
+    ):
+        if event.asset != EthereumToken(tx_log.address):
+            return True
+        amount_raw = hex_or_bytes_to_int(tx_log.data)
+        amount = asset_normalized_value(amount=amount_raw, asset=event.asset)
+        if event.balance.amount == amount:  # noqa: E501
+            event.event_type = HistoryEventType.STAKING
+            event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+            event.counterparty = 'pickle finance'
+            event.notes = f'Return {event.balance.amount} {event.asset.symbol} to the pickle contract'  # noqa: E501
+    elif (
```

Comment:
why no comment here?

Reply:
I just got them in deposit/withdraw and by the code you can clearly see if it is give the wrapped token or receive it. But I can add more comments if is not clear

Reply:
Either put a comment in all or in none